src/server/mod.rs

```diff
@@ -0,0 +1,1249 @@
+                            None => false,
+                        }).map(|x| x.0 ).collect();
+                        if potential_imports.is_empty() {
+                            return None;
+                        }
+
+                        let inner_actions: Vec<lsp::CodeActionOrCommand> = potential_imports.iter().map(|package_name| {
+                            lsp::CodeAction {
+                                title: format!("Import `{}`", package_name),
+                                kind: Some(lsp::CodeActionKind::QUICKFIX),
+                                diagnostics: None,
+                                edit: Some(lsp::WorkspaceEdit {
+                                    changes: Some(HashMap::from([
+                                        (params.text_document.uri.clone(), vec![
+                                            lsp::TextEdit {
+                                                range: lsp::Range {
+                                                    start: import_position,
+                                                    end: import_position,
+                                                },
+                                                new_text: format!("import \"{}\"\n", package_name),
+                                            }
+                                        ])
+                                    ])),
+                                    document_changes: None,
+                                    change_annotations: None,
+                                }),
+                                command: None,
+                                is_preferred: Some(true),
+                                disabled: None,
+                                data: None,
+                            }.into()
+                        }).collect();
+                        return Some(inner_actions);
+                    },
+                    _ => return None,
+                }
+            }
+            None
+        }).filter(|action| action.is_some()).map(|action| {
+            action.expect("Previous .filter() call failed.")
+        }).flatten().collect();
+
+        return Ok(Some(actions));
+    }
+}
+
+#[derive(Default, Clone)]
+struct NodeFinderResult<'a> {
+    node: Option<flux::semantic::walk::Node<'a>>,
+    path: Vec<flux::semantic::walk::Node<'a>>,
+}
+
+fn find_node(
+    node: flux::semantic::walk::Node<'_>,
+    position: lsp::Position,
+) -> NodeFinderResult<'_> {
+    let mut result = NodeFinderResult::default();
+    let mut visitor = semantic::NodeFinderVisitor::new(position);
+
+    flux::semantic::walk::walk(&mut visitor, node);
+
+    result.node = visitor.node;
+    result.path = visitor.path;
+
+    result
+}
+
+// Url::to_file_path doesn't exist in wasm-unknown-unknown, for kinda
+// obvious reasons. Ignore these tests when executing against that target.
+#[cfg(all(test, not(target_arch = "wasm32")))]
+#[allow(deprecated, clippy::panic, clippy::unwrap_used)]
```

Comment:
I might move this attribute to be module-level _inside the new test module_.

Reply:
Yeah, good catch.